write-fonts/src/tables/variations.rs

```diff
@@ -21,6 +21,237 @@ impl VariationRegionList {
     }
 }
 
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats#packed-point-numbers>
+pub struct PackedPointNumbers {
+    is_all: bool,
+    numbers: Vec<u16>,
+}
+
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats#packed-deltas>
+pub struct PackedDeltas {
+    deltas: Vec<i16>,
+}
+
+impl Validate for PackedDeltas {
+    fn validate_impl(&self, _ctx: &mut ValidationCtx) {}
+}
+
+impl FontWrite for PackedDeltas {
+    fn write_into(&self, writer: &mut TableWriter) {
+        for run in self.iter_runs() {
+            run.write_into(writer)
+        }
+    }
+}
+
+impl PackedDeltas {
+    /// Construct a `PackedDeltas` from a vector of raw delta values.
+    pub fn new(deltas: Vec<i16>) -> Self {
+        Self { deltas }
+    }
+
+    fn iter_runs(&self) -> impl Iterator<Item = PackedDeltaRun> {
+        const MAX_POINTS_PER_RUN: usize = 64;
+
+        fn in_i8_range(val: i16) -> bool {
+            const MIN: i16 = i8::MIN as i16;
+            const MAX: i16 = i8::MAX as i16;
+            (MIN..=MAX).contains(&val)
+        }
+
+        fn count_leading_zeros(slice: &[i16]) -> usize {
+            slice.iter().position(|x| *x != 0).unwrap_or(slice.len())
+        }
+
+        /// compute the number of deltas in the next run, and whether they are i8s or not
+        fn next_run_len(slice: &[i16]) -> (usize, bool) {
+            let (first, rest) = slice.split_first().expect("bounds checked before here");
+            debug_assert!(*first != 0);
+            let is_1_byte = in_i8_range(*first);
+
+            for (i, raw) in rest.iter().copied().enumerate().take(MAX_POINTS_PER_RUN) {
+                let two_zeros = raw == 0 && rest.get(i + 1) == Some(&0);
+                let different_enc_len = in_i8_range(raw) != is_1_byte;
+                if two_zeros || different_enc_len {
+                    return (i + 1, is_1_byte);
+                }
+            }
+
+            (slice.len(), is_1_byte)
+        }
+
+        let mut deltas = self.deltas.as_slice();
+
+        std::iter::from_fn(move || {
```

Comment:
I still find this pattern harder to read than a struct plus impl iter.

Reply:
I agree, and I prefer a manual impl when the logic is more complicated, but in these simple cases I really appreciate the concision of `from_fn`.